apps/vaporgui/GeometryWidget.cpp

```diff
@@ -462,6 +472,41 @@ void GeometryWidget::Update(ParamsMgr *paramsMgr,
     }
 }
 
+void GeometryWidget::_resetBoxIfNeeded(
+    const std::vector<double> &minFullExt,
+    const std::vector<double> &maxFullExt
+) {
+    std::vector<double> boxMin, boxMax;
+    _box->GetExtents( boxMin, boxMax );
+  
+    // If the box extents are of a different dimensionality than the extents
+    // of the currently active variable, reset the box to the variable extents 
+    //
+    if ( boxMin.size() != minFullExt.size() ||
+         boxMax.size() != maxFullExt.size() 
+    ) {
+        _box->SetExtents( minFullExt, maxFullExt );
+        return;
+    }
+
+    // Now check to see if the box is entirely outside of the variable's extents 
+    //
+    for ( int i=0; i<minFullExt.size(); i++ ) {
```

Comment:
@sgpearse This algorithm is not quite right. It will reset the box if *any* of the axis fail to overlap. I believe we want to reset only when *all* of the axis fail to overlap. See https://stackoverflow.com/questions/20925818/algorithm-to-check-if-two-boxes-overlap

How about we add a BoxIntersect() method to the Box class? This would be a useful general utility.

Reply:
@clyne Hmm, I do believe that the current algorithm is correct.  

In this bug, only the Z axis of TSLB and the other 3D variables fail to overlap.  However the X and Y axes do indeed overlap.  If we require all 3 to fail in overlapping, we miss the case reported in #1375.

Reply:
You are correct, but the code is confusing. Checking boxMin < minFullExt and boxMax > maxFullExt are irrelevant. min is always less than or equal to max. So you simply need:

```
if ( boxMax[i] < minFullExt[i] II boxMin[i] > minFullExt[i]) {
  _box->SetExtents( minFullExt, maxFullExt );
  return;
}
```